### ISSUE TITLE
SRM & Loihi Neural Dynamics, Parameters? 

### ISSUE THREAD

**Issue by zenmar** [bamsumit/slayerPytorch#79]
Hi,

I was wondering if there was some way to replicate the SRM neural behaviour with the Loihi parameters?

Training a couple of SNNs using the SRM model on MNIST/F-MNIST I got to 95%+ accuracy; but trying to do the same with the Loihi model the learning happens at much much slower rate (in term of epochs).

Where, with the SRM model accuracy goes above 90% very quickly, after ~3-5 epochs (depending on coding); with the Loihi neuron-model (and the 'default' parameters, the ones you're using for NMNIST in the tutorial), it's only residing at around 50-60%.

Regarding SLAYERAuto, I'm still not 100% sure about its syntax.

I have this network:

> cnnNetwork(
>   (slayer): spikeLayer()
>   (conv1): _convLayer(1, 16, kernel_size=(5, 5, 1), stride=(1, 1, 1), padding=(1, 1, 0), bias=False)
>   (conv2): _convLayer(16, 32, kernel_size=(3, 3, 1), stride=(1, 1, 1), padding=(1, 1, 0), bias=False)
>   (conv3): _convLayer(32, 64, kernel_size=(3, 3, 1), stride=(1, 1, 1), padding=(1, 1, 0), bias=False)
>   (pool1): _poolLayer(1, 1, kernel_size=(2, 2, 1), stride=(2, 2, 1), bias=False)
>   (pool2): _poolLayer(1, 1, kernel_size=(2, 2, 1), stride=(2, 2, 1), bias=False)
>   (fc1): _denseLayer(64, 10, kernel_size=(8, 8, 1), stride=(1, 1, 1), bias=False)
> )

When trying to rewrite it with SLAYERAuto like this:
>  'layer' : [
>         {'dim': '34x34x1'},
>         {'dim': '16c5z'},
>         {'dim': '2a'},
>         {'dim': '32c3'},
>         {'dim': '2a'},
>         {'dim': '64c3'},
>         {'dim': 10}
>     ],

I end up with this:

> Network(
>   (slayer): spikeLayer()
>   (blocks): ModuleList(
>     (0): convBlock(
>       (slayer): spikeLayer()
>       (weightOp): _convLayer(1, 16, kernel_size=(5, 5, 1), stride=(1, 1, 1), **_padding=(2, 2, 0),_** bias=False)
>     )
>     (1): poolBlock(
>       (slayer): spikeLayer()
>       (weightOp): _poolLayer(1, 1, kernel_size=(2, 2, 1), stride=(2, 2, 1), bias=False)
>     )
>     (2): convBlock(
>       (slayer): spikeLayer()
>       (weightOp): _convLayer(16, 32, kernel_size=(3, 3, 1), stride=(1, 1, 1), padding=(1, 1, 0), bias=False)
>     )
>     (3): poolBlock(
>       (slayer): spikeLayer()
>       (weightOp): _poolLayer(1, 1, kernel_size=(2, 2, 1), stride=(2, 2, 1), bias=False)
>     )
>     (4): convBlock(
>       (slayer): spikeLayer()
>       (weightOp): _convLayer(32, 64, kernel_size=(3, 3, 1), stride=(1, 1, 1), padding=(1, 1, 0), bias=False)
>     )
>     (5): flattenBlock()
>     (6): denseBlock(
>       (slayer): spikeLayer()
>       (weightOp): _denseLayer(5184, 10, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
>     )
>   )
> )

(The layers are conv1 -> pool1 -> conv2 -> pool2 -> conv3 -> dense)

Especially, in the first convlayer, the padding is not the same (which might impact the learning a bit?).

Thanks!

**Comment by bamsumit** [comment]
You may start by making the time constants in similar range. The hyperparameters are going to be different because the neuron model and dynamics are different. In addition Loihi model is a fixed precision model. So your convergence is going to be slow.

On padding, just add a `pad` field in the layer description:
```
'layer' : [
        {'dim': '34x34x1'},
        {'dim': '16c5z', 'pad': 1},
        {'dim': '2a'},
        {'dim': '32c3'},
        {'dim': '2a'},
        {'dim': '64c3'},
        {'dim': 10}
],
```

**Comment by zenmar** [comment]
Thanks for the reply!

Adding the 'pad' field, doesn't seem to work though, when calling the assistant to build the net using I'm still gettting:

```Network Architecture:  \
|   Type   |  W  |  H  |  C  | ker | str | pad |delay|  params  |  \
|Input     |   34|   34|    1|     |     |     |False|          |  
|Conv      |   34|   34|   16|    5|    1|    2|False|       400|  
|Pool      |   17|   17|   16|    2|     |     |False|          |  
|Conv      |   17|   17|   32|    3|    1|    1|False|      4608|  
|Pool      |    9|    9|   32|    2|     |     |False|          |  
|Conv      |    9|    9|   64|    3|    1|    1|False|     18432|  
|Dense     |    1|    1|   10|     |     |     |False|     51840|  
|Total                                               |     75280|  
TODO core usage estimator  
Max PSP kernel: 195.0  
Scaling neuron[scaleRho] by Max PSP Kernel @slayerLoihi```

**Comment by bamsumit** [comment]
Whatever is in the `neuron` field needs to be tuned.
